File: DataContract/UserDTO.cs
namespace DataContract
{
    public class UserDTO
    {
        public int? Id { get; set; }
        public string UserName { get; set; }
        public string Email { get; set; }
        public DateTimeOffset DateOfBirth { get; set; }
        public int Age { get { return (DateTime.Today - DateOfBirth).Days / 365; } }
        public DateTimeOffset? LastActive { get; set; }
        public DateTimeOffset? CreatedAt { get; set; }
        public string Gender { get; set; }
        public bool IsInactive { get; set; }
        public string? Bio { get; set; }
        public string? Interests { get; set; }
        public string? City { get; set; }
        public string? Country { get; set; }
        public string? ProfileImageUrl { get; set; }
        public string? ProfileImagePublicId { get; set; }
        public List<PhotoDTO> Photos { get; set; } = new();
        public List<int> FollowedUserIds { get; set; } = new();
        public List<int> FollowedByUserIds { get; set; } = new();
    }
}

File: SocialSphere.UI/src/app/core/interfaces/user-dto.ts
import { PhotoDTO } from './photo-dto';

export interface UserDTO {
  id?: number;
  userName: string;
  email: string;
  dateOfBirth: Date;
  age?: number;
  lastActive?: Date;
  createdAt?: Date;
  gender: string;
  isInactive?: boolean;
  bio: string;
  interests: string;
  city: string;
  country: string;
  profileImageUrl?: string;
  profileImagePublicId?: string;
  photos?: PhotoDTO[];
  followedUserIds?: number[]; // users who i follow
  followedByUserIds?: number[]; // users who follow me
}
